src/thunderbots/software/util/canvas_messenger/canvas_messenger.cpp

```diff
@@ -18,42 +18,57 @@ void CanvasMessenger::initializePublisher(ros::NodeHandle node_handle)
         Util::Constants::VISUALIZER_DRAW_LAYER_TOPIC, BUFFER_SIZE);
 }
 
-void CanvasMessenger::publishAndClearAllLayers()
+void CanvasMessenger::publishAndClearLayer(Layer layer)
 {
     // Take ownership of the layers for the duration of this function
-    std::lock_guard<std::mutex> best_known_pass_lock(layers_map_lock);
-
-    // Limit rate of the message publishing
-    // Get the time right now
-    const std::chrono::time_point<std::chrono::system_clock> now =
-        std::chrono::system_clock::now();
-    const int64_t elapsed_ns =
-        std::chrono::duration_cast<std::chrono::nanoseconds>(now - time_last_published)
-            .count();
-    const double elapsed_ms = elapsed_ns / 1.0e6;
-
-    // Do not do anything if the time passed hasn't been
-    // long enough
-    if (elapsed_ms < DESIRED_PERIOD_MS)
-        return;
-
-    // Send a payload per layer of messages
-    for (const auto& layer_pair : this->layers_map)
+    std::lock_guard<std::mutex> layers_map_lock(layers_map_mutex);
+
+    //    // Limit rate of the message publishing
```

Comment:
This should be implemented or the commented out code should be removed

Reply:
Implemented!